server/v2/api/rest/server.go

```diff
@@ -0,0 +1,89 @@
+package rest
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"cosmossdk.io/core/transaction"
+	"cosmossdk.io/log"
+	serverv2 "cosmossdk.io/server/v2"
+	"cosmossdk.io/server/v2/appmanager"
+)
+
+const (
+	ServerName = "rest-v2"
+)
+
+type Server[T transaction.Tx] struct {
+	logger log.Logger
+	router *http.ServeMux
+
+	httpServer *http.Server
+	config     *Config
+	cfgOptions []CfgOption
+}
+
+func New[T transaction.Tx](cfgOptions ...CfgOption) *Server[T] {
+	return &Server[T]{
+		cfgOptions: cfgOptions,
+	}
+}
+
+func (s *Server[T]) Name() string {
+	return ServerName
+}
+
+func (s *Server[T]) Init(appI serverv2.AppI[T], cfg map[string]any, logger log.Logger) error {
+	s.logger = logger.With(log.ModuleKey, s.Name())
+
+	s.config = s.Config().(*Config)
+
+	var appManager *appmanager.AppManager[T]
+	appManager = appI.GetAppManager()
+
+	s.router = http.NewServeMux()
+	s.router.Handle("/", NewDefaultHandler(appManager))
+
+	return nil
+}
+
+func (s *Server[T]) Start(ctx context.Context) error {
+	s.httpServer = &http.Server{
+		Addr:    s.config.Address,
+		Handler: s.router,
+	}
+
+	go func() {
```

Comment:
no need to be in a go routine, we can be blocking, server components are already running in  a go routine